pkg/comp-functions/runtime/function_mgr.go

```diff
@@ -475,6 +475,26 @@ func (s *ServiceRuntime) SetDesiredKubeObjectWithName(obj client.Object, objectN
 	return s.SetDesiredComposedResourceWithName(kobj, resourceName)
 }
 
+// KubeOptionLabeler adds the given labels to the kube object.
+func KubeOptionAddLabels(labels map[string]string) KubeObjectOption {
+	return func(obj *xkube.Object) {
+		current := obj.GetLabels()
+		if current == nil {
+			current = map[string]string{}
+		}
+		for val, key := range labels {
+			current[val] = key
+		}
+		obj.SetLabels(current)
+	}
+}
+
+// KubeOptionObserveOnly sets the object to only observe.
+func KubeOptionObserveOnly(obj *xkube.Object) {
```

Comment:
There's already a function that does just that `KubeOptionObserve(obj *xkube.Object)`

Reply:
I've removed mine